debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java

```diff
@@ -349,6 +351,31 @@ public ReplicationStream startStreaming(Lsn offset, WalPositionLocator walPositi
         }
     }
 
+    protected void validateSlotIsInExpectedState(Lsn lsn) throws SQLException {
+        try (Statement stmt = pgConnection().createStatement()) {
+            String seekCommand = String.format(
+                    "SELECT pg_replication_slot_advance('%s', '%s')",
+                    slotName,
+                    lsn.asString());
+            LOGGER.info("Seeking to {} on the replication slot with command {}", lsn, seekCommand);
+            stmt.execute(seekCommand);
+        }
+        catch (PSQLException e) {
+            if (e.getMessage().matches("ERROR: function pg_replication_slot_advance.*does not exist(.|\\n)*")) {
+                LOGGER.info("Postgres server doesn't support the command pg_replication_slot_advance(). Not seeking to last known offset.");
+            }
+            else if (e.getMessage().matches("ERROR: cannot advance replication slot to.*")) {
+                throw new DebeziumException(
+                        String.format("Cannot seek to the last known offset '%s' on replication slot '%s'. Error from server: %s", lsn.asString(), slotName,
+                                e.getMessage()),
+                        e);
+            }
+            else {
+                throw e;
```

Comment:
@jpechane I believe the Debezium connector 2.0 treats PSQLException as retriable  exception, so this might trigger continuous restart of the connector instead of stopping it.